packages/calypso-e2e/src/lib/pages/plans-page.ts

```diff
@@ -89,6 +95,21 @@ export class PlansPage {
 		await locator.first().click();
 	}
 
+	/**
+	 * Selects the plan on the modal upsell.
+	 *
+	 * @param {Plans} plan Plan to select.
+	 */
+	async selectModalUpsellPlan( plan: Plans ): Promise< void > {
+		if ( plan !== 'Free' && plan !== 'Personal' ) {
+			throw Error( `Unsupported plan to be selected in modal upsell: ${ plan }` );
+		}
+
+		const locator = this.page.locator( selectors.selectModalUpsellPlanButton( plan ) );
+
+		await locator.first().click();
+	}
+
```

Comment:
**issue:blocking**- strictly speaking, this upsell modal is not part of the PlansPage, but only for the `SignupPickPlanPage`, so it shouldn't be here.

Reply:
But the selectors need to live in `PlansPage` right?